Source/Core/VideoCommon/AVIDump.cpp

```diff
@@ -95,43 +96,48 @@ bool AVIDump::Start(int w, int h)
 bool AVIDump::CreateVideoFile()
 {
   const char* s_format = g_Config.sDumpFormat.c_str();
-
-  AVCodec* codec = nullptr;
-
-  std::stringstream file_ss;
-  file_ss << File::GetUserPath(D_DUMPFRAMES_IDX)
-                   << "framedump" << s_file_index
-                   << "." << s_format;
-  std::string filename = file_ss.str();
-  File::CreateFullPath(filename);
-
-  // Ask to delete file
-  if (File::Exists(filename))
+  
+  std::string s_dump_path = g_Config.sDumpURL;
+  
+  if (s_dump_path.empty())
   {
-    if (SConfig::GetInstance().m_DumpFramesSilent ||
-        AskYesNoT("Delete the existing file '%s'?", filename.c_str()))
-    {
-      File::Delete(filename);
-    }
-    else
+    std::stringstream file_ss;
```

Comment:
This block can be refactored into another function to make this one more short and clear. Something like `std::pair<bool, std::string> GetDefaultDumpPath()`?